File: ECommerceAPI/ECommerceAPI/Models/Product.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace ECommerceAPI.Models
{
    public class Product
    {
        public int ProductId { get; set; }
        [Required(ErrorMessage = "name_required")]
        public string Name { get; set; }
        [Required(ErrorMessage = "code_required")]
        public string Code { get; set; }
        public int ProductTypeId { get; set; }
        public bool Active { get; set; }
        public ProductType ProductType { get; set; }
        public List<Image> Images { get; set; }
        public List<Information> Informations { get; set; }
        public Product()
        {
            Images = new List<Image>();
            Informations = new List<Information>();
            ProductType = new ProductType();
        }
    }
}

File: ECommerceFront/src/app/models/product.ts
import { Information } from './information';
import { ProductType } from './productType';
import { Image } from './image';
import { UploadFileInfo } from './uploadFileInfo';

export class Product {
    productId: number;
    name: string;
    code: string;
    productType: ProductType;
    active: boolean;
    informations: Array<Information>;
    images: Array<Image>;

    constructor() {
        this.name = '';
        this.code = '';
        this.informations = new Array<Information>();
        this.active = false;
        this.images = new Array<Image>();
        this.productType = new ProductType();
    }

    asignProduct(product) {
        if (!product) {
            return;
        }
        this.name = product.name ? product.name : this.name;
        this.code = product.code ? product.code : this.code;
        if (product.productType && product.productType.productTypeId > 0) {
            this.productType.productTypeId = +product.productType.productTypeId;
        }
        this.active = product.active ? product.active : this.active;
        this.informations = product.informations.length > 0 ? product.informations : this.informations;
        this.images = product.images.length ? product.images : this.images;
    }

    getFormData(): FormData {
        const formData = new FormData();
        this.images.forEach(img => formData.append(img.name, img.image));
        return formData;
    }

    addImageNames(imageInfo: Array<UploadFileInfo>) {
        this.images.forEach(img => {
            const imgInfo = imageInfo.find(i => i.pastName === img.name);
            if (imgInfo) {
                img.name = imgInfo.currentName;
                img.path = null;
            }
        });
    }
}
